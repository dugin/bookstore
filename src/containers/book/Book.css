.Book {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 410px;
  border: 1px solid #d1d1d1;
  margin-top: -1px;
  background-color: #fff; }
  .Book:hover {
    box-shadow: 0 15px 32px 0 rgba(0, 0, 0, 0.15); }
  .Book__figureContainer {
    margin-top: 15px;
    text-align: center;
    border-bottom: 1px solid #e8e8e8;
    min-height: 290px; }
    .Book__figureContainer__discountTag {
      border-radius: 50%;
      border: 2px solid #000;
      position: absolute;
      left: 10px;
      width: 40px;
      font-weight: bold;
      font-size: 12px;
      padding: 5px;
      height: 40px;
      line-height: 14px; }
    .Book__figureContainer__img {
      height: 120px; }
    .Book__figureContainer__caption__name {
      margin-top: 10px;
      font-size: 15px;
      margin-bottom: 5px; }
    .Book__figureContainer__caption__subtitle {
      font-size: 13px;
      margin-top: 0;
      color: #6d6d6d; }
    .Book__figureContainer__caption__description {
      font-size: 11px;
      color: #939393;
      text-align: justify;
      font-weight: 300;
      overflow: hidden;
      text-overflow: ellipsis;
      display: -webkit-box;
      /*! autoprefixer: off */
      -webkit-box-orient: vertical;
      /*! autoprefixer: on */
      -webkit-line-clamp: 4; }
  .Book__valueContainer__discount {
    margin-right: 10px;
    font-size: 14px;
    text-decoration: line-through !important;
    color: #939393; }
  .Book__valueContainer__price {
    font-size: 16px;
    font-weight: bold; }
  .Book__actions {
    display: flex;
    margin-top: 15px;
    width: 100%;
    justify-content: space-between; }
    .Book__actions__addToCart, .Book__alert__container__goToCart--btn {
      background-color: #c33fa3;
      border: 1px solid #c33fa3;
      border-radius: 5px;
      color: #fff;
      font-size: 12px; }
      .Book__actions__addToCart:hover, .Book__alert__container__goToCart--btn:hover {
        background-color: #9e3183; }
    .Book__actions__quantityContainer {
      margin-right: 10px;
      display: flex;
      justify-content: center;
      align-items: center; }
      .Book__actions__quantityContainer__quantity {
        margin: 0 10px;
        min-width: 18px;
        text-align: center; }
      .Book__actions__quantityContainer__minus, .Book__actions__quantityContainer__plus {
        display: flex;
        align-items: center;
        padding: 4px 5px;
        background-color: white;
        border: 1px solid #939393;
        border-radius: 5px;
        width: 30px;
        color: #939393; }
        .Book__actions__quantityContainer__minus .material-icons, .Book__actions__quantityContainer__plus .material-icons {
          font-size: 18px;
          position: relative; }
        .Book__actions__quantityContainer__minus:hover, .Book__actions__quantityContainer__plus:hover {
          border: 1px solid #c33fa3;
          color: #c33fa3; }
  .Book__alert {
    position: absolute;
    z-index: 1;
    height: 100%;
    width: 100%;
    padding: 20px;
    background-color: rgba(0, 0, 0, 0.8);
    text-align: center; }
    .Book__alert__container {
      color: #fff;
      font-size: 20px;
      height: 100%;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center; }
      .Book__alert__container__icon {
        font-size: 42px;
        margin-bottom: 10px; }
      .Book__alert__container__goToCart--btn {
        margin-top: 35px;
        font-size: 14px;
        width: 100%;
        padding: 5px 10px; }
      .Book__alert__container__pickMore--btn {
        margin-top: 10px;
        background-color: transparent;
        border: none;
        font-size: 12px;
        text-decoration: underline;
        color: #fff; }
