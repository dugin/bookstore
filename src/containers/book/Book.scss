@import "../../variables";

.Book {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 370px;
  border: 1px solid $color__lineCard--gray;
  margin-top: -1px;
  background-color: $color--white;

  &:hover {
    box-shadow: 0 15px 32px 0 rgba(0, 0, 0, 0.15);
  }

  &__figureContainer {
    margin-top: 15px;
    text-align: center;
    border-bottom: 1px solid $color__line--gray;
    min-height: 250px;

    &__discountTag {
      border-radius: 50%;
      border: 2px solid #000;
      position: absolute;
      left: 10px;
      width: 40px;
      font-weight: bold;
      font-size: 12px;
      padding: 5px;
      height: 40px;
      line-height: 14px;
    }

    &__img {
      height: 120px;
    }

    &__caption {

      &__name {
        margin-top: 10px;
        font-size: 15px;
      }

      &__description {
        font-size: 11px;
        color: $color__text--gray;
        text-align: justify;
        font-weight: 300;
        overflow: hidden;
        text-overflow: ellipsis;
        display: -webkit-box;
        /*! autoprefixer: off */
        -webkit-box-orient: vertical;
        /*! autoprefixer: on */
        -webkit-line-clamp: 4;
      }
    }
  }

  &__valueContainer {

    &__discount {
      margin-right: 10px;
      font-size: 14px;
      text-decoration: line-through !important;
      color: $color__text--gray;
    }
    &__price {
      font-size: 16px;
      font-weight: bold;
    }
  }

  &__actions {
    display: flex;
    margin-top: 15px;
    width: 100%;
    justify-content: space-between;

    &__addToCart {
      background-color: $color--accent;
      border: 1px solid $color--accent;
      border-radius: 5px;
      color: #fff;
      font-size: 12px;

      &:hover {
        background-color: darken($color--accent, 10);
      }

    }

    &__quantityContainer {
      margin-right: 10px;
      display: flex;
      justify-content: center;
      align-items: center;

      &__quantity {
        margin: 0 10px;
      }

      &__minus, &__plus {
        font-size: 12px;
        padding: 4px 5px;
        background-color: white;
        border: 1px solid $color__text--gray;
        border-radius: 5px;
        width: 30px;

        color: $color__text--gray;;

        &:hover {
          border: 1px solid $color--accent;
          color: $color--accent;
        }

      }

    }
  }
}