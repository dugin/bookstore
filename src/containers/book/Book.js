import React, {Component} from 'react';
import {connect} from 'react-redux';
import './Book.css';
import NumberFormat from 'react-number-format'

export class Book extends Component {

    render() {
        return (
            <div className="Book container">
                <figure className="Book__figureContainer">
                    <img src={this.props.book.imageLinks.thumbnail} className="Book__figureContainer__img" alt=""/>
                    <figcaption className="Book__figureContainer__caption">
                        <h5 className="Book__figureContainer__caption__name"> {this.props.book.title}</h5>
                        <p className="Book__figureContainer__caption__description"> {this.props.book.description}</p>
                    </figcaption>
                </figure>

                <section className="Book__valueContainer">

                    <NumberFormat className="Book__valueContainer__price" value={this.props.book.price}
                                  decimalPrecision={2} displayType={'text'} decimalSeparator=',' prefix={'R$'}/>


                </section>

                <section className="Book__actions">
                    <div className="Book__actions__quantityContainer">
                        <button className="Book__actions__quantityContainer__minus"> - </button>
                        <span className="Book__actions__quantityContainer__quantity"> 1 </span>
                        <button className="Book__actions__quantityContainer__plus "> + </button>
                    </div>
                    <button className="Book__actions__addToCart">
                        Adicionar ao Carrinho
                    </button>
                </section>

            </div>
        )
    }
}

// const mapStateToProps = () => {
//     return {
//         books: state
//     }
// }

export default Book;