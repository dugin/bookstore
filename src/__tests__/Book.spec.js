import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Book} from "../containers/book/Book";
import {props} from "../__mocks__/bookMock";

describe('Book', () => {

    const book = shallow(<Book {...props}/>);

    it('renders properly', () => {

        expect(toJson(book)).toMatchSnapshot();

    });

    it('renders its image, name, description, price', () => {
        expect(book.find('.Book__figureContainer__img').exists()).toBe(true);
        expect(book.find('.Book__figureContainer__caption__name').exists()).toBe(true);
        expect(book.find('.Book__figureContainer__caption__description').exists()).toBe(true);
        expect(book.find('.Book__valueContainer__price').exists()).toBe(true);

    });

    it('renders discount price and tag if exists', () => {
        expect(book.find('.Book__valueContainer__discount').exists()).toBe(true);
        expect(book.find('.Book__figureContainer__discountTag').exists()).toBe(true);
    });

    it('renders quantity and `Adicionar ao Carrinho` button ', () => {
        expect(book.find('.Book__actions__quantityContainer__minus').exists()).toBe(true);
        expect(book.find('.Book__actions__quantityContainer__quantity').exists()).toBe(true);
        expect(book.find('.Book__actions__quantityContainer__plus').exists()).toBe(true);
        expect(book.find('.Book__actions__addToCart').text()).toBe('Adicionar ao Carrinho');
    });

    describe('Add/Subtract buttons', () => {

        const plusBtn = book.find('.Book__actions__quantityContainer__plus');
        const minusBtn = book.find('.Book__actions__quantityContainer__minus');

        beforeEach(() => {
            book.setState({quantity: 1});
        });

        it('adds one on plus button click ', () => {
            expect(plusBtn.length).toEqual(1);
            plusBtn.simulate('click');

            expect(book.state().quantity).toEqual(2);
        });

        it('do not subtract on minus button if quantity == 1', () => {
            expect(plusBtn.length).toEqual(1);
            expect(minusBtn.length).toEqual(1);

            minusBtn.simulate('click');

            expect(book.state().quantity).toEqual(1);
        });

        it('subtract one on minus button click if quantity > 1', () => {
            book.setState({quantity: 3});

            expect(minusBtn.length).toEqual(1);
            minusBtn.simulate('click');

            expect(book.state().quantity).toEqual(2);
        });
    });
});