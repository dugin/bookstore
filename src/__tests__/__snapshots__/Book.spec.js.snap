// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book renders properly 1`] = `
<Book
  addToCart={[Function]}
  amount={2}
  book={
    Object {
      "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
      "discount": 5,
      "id": "nggnmAEACAAJ",
      "imageLinks": Object {
        "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
        "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
      },
      "price": 45,
      "title": "The Linux Command Line",
    }
  }
>
  <div
    className="Book container"
  >
    <figure
      className="Book__figureContainer"
    >
      <div
        className="Book__figureContainer__discountTag"
      >
        12% OFF
      </div>
      <BookInfo
        addToCart={[Function]}
        amount={2}
        book={
          Object {
            "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
            "discount": 5,
            "id": "nggnmAEACAAJ",
            "imageLinks": Object {
              "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
              "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
            },
            "price": 45,
            "title": "The Linux Command Line",
          }
        }
      >
        <div
          className="Book__wrapper"
        >
          <img
            alt=""
            className="Book__figureContainer__img"
            src="http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
          />
          <figcaption
            className="Book__figureContainer__caption"
          >
            <h5
              className="Book__figureContainer__caption__name"
            >
               
              The Linux Command Line
            </h5>
            <p
              className="Book__figureContainer__caption__subtitle"
            />
            <p
              className="Book__figureContainer__caption__description"
            >
               
              You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial "shell shock," you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's "Evolution of a SysAdmin"
            </p>
          </figcaption>
        </div>
      </BookInfo>
    </figure>
    <section
      className="Book__valueContainer"
    >
      <NumberFormat
        allowNegative={true}
        className="Book__valueContainer__discount"
        decimalPrecision={2}
        decimalSeparator=","
        displayType="text"
        isAllowed={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        prefix="R$ "
        suffix=""
        type="text"
        value={45}
      >
        <span
          className="Book__valueContainer__discount"
        >
          R$ 45,00
        </span>
      </NumberFormat>
      <NumberFormat
        allowNegative={true}
        className="Book__valueContainer__price"
        decimalPrecision={2}
        decimalSeparator=","
        displayType="text"
        isAllowed={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        prefix="R$ "
        suffix=""
        type="text"
        value={40}
      >
        <span
          className="Book__valueContainer__price"
        >
          R$ 40,00
        </span>
      </NumberFormat>
    </section>
    <section
      className="Book__actions"
    >
      <QuantityButtons
        onAddOrRemove={[Function]}
      >
        <div
          className="Book__actions__quantityContainer"
        >
          <button
            className="Book__actions__quantityContainer__minus"
            onClick={[Function]}
          >
            <i
              className="material-icons"
            >
              remove
            </i>
          </button>
          <span
            className="Book__actions__quantityContainer__quantity"
          >
             
            1
             
          </span>
          <button
            className="Book__actions__quantityContainer__plus "
            onClick={[Function]}
          >
            <i
              className="material-icons"
            >
              add
            </i>
          </button>
        </div>
      </QuantityButtons>
      <button
        className="Book__actions__addToCart"
        onClick={[Function]}
      >
        Adicionar ao Carrinho
      </button>
    </section>
  </div>
</Book>
`;
